### PR TITLE
Setup typescript

### PR DESCRIPTION
**Pull request by EscapedGibbon**
First draft

### CODE REVIEW

File: src/KMeansResult.ts [hunk 1]
```diff
@@ -2,17 +2,29 @@ import { updateClusterID } from './utils';
...
+  centroid: number;
+  error: number | null;
+  size: number;
+}
 export default class KMeansResult {
   /**
    * Result of the kmeans algorithm
    * @param {Array<number>} clusters - the cluster identifier for each data dot
-   * @param {Array<Array<object>>} centroids - the K centers in format [x,y,z,...], the error and size of the cluster
+   * @param {Array<Centroid>} centroids - the K centers in format [x,y,z,...], the error and size of the cluster
```
Comment (stropitek) [inline src/KMeansResult.ts:14]: As for distance, the jsdoc type annotations can be removed

File: src/KMeansResult.ts [hunk 2]
```diff
@@ -39,10 +51,10 @@ export default class KMeansResult {
    * @param {Array<Array<number>>} data - the [x,y,z,...] points to cluster
    * @return {KMeansResult}
    */
-  computeInformation(data) {
-    let enrichedCentroids = this.centroids.map((centroid) => {
+  computeInformation(data: Array<Array<number>>): KMeansResult {
```
Comment (stropitek) [inline src/KMeansResult.ts:54]: With simple types, I find the other syntax for arrays easier to read
```suggestion
  computeInformation(data: number[][]): KMeansResult {
```

**Comment by stropitek** [comment]
@EscapedGibbon can you ping when you are done working on what we discussed about this PR?

**Comment by EscapedGibbon** [comment]
@stropitek I think I am done with it. The default exports were removed and I think documentation should look better once it is built. But there is a coverage test that doesn't go through.

**Review by stropitek — APPROVED** [review]

**Pull request merged**